stuff_to_do for redmine 2.0 with rails 3 (Feature #6551)

neffets opened the pull request:
removed deprecated ARcondition, added compatibility for rails-2 and rails-3, so that stuff-to-do plugin now is usable for redmine 2.0 too; tested in redmine 1.4-stable and redmine 2.0-trunk.

See: https://projects.littlestreamsoftware.com/issues/6551

neffets committed:
fixes #6551 @3h - removed deprecated ARcondition, added compatibility for rails-2 and rails-3, so that stuff-to-do plugin now is usable for redmine 2.0 too; tested in redmine 1.4-stable and redmine 2.0-trunk

alevkin committed:
Fix "PG::Error: ERROR:  operator does not exist: boolean = integer"

Commit:
Fixes for latest ruby/rails

Commit:
Remove old shit

Commit:
Add new shit

Commit:
For New Redmine

Commit:
For New Redmine

Commit:
fixes

Commit:
estimate time display upd

Commit:
menu fix for current redmine

ande3577 committed:
Closes #4: Fix viewing other's stuff to do

ande3577 committed:
Closes #1: Add stuff to do permission.

commit f2626ef91a1334e0ced24fbff4182f331af91285
Author: David S Anderson <ande3577@gmail.com>
Date:   Sun Sep 2 15:38:43 2012 -0500

    Added permissions for stuff to do plugin

ande3577 committed:
Closes #5: added user option to enable/disable stuff to do plugin

ande3577 committed:
Closes #8: No longer shows stuff to do enabled if user does not have permission

ande3577 committed:
Closes #11: Time grid no longer creates error

ande3577 committed:
Closes #6: Can add to stuff to do from issue page or context menu

ande3577 committed:
Added not about version compatibility

ande3577 committed:
Closes #16: changed account menu to use a hook

ande3577 committed:
changed default enabled setting to true

Rumega committed:
Added code to support ActionMailer from Rails 3 (this fixes an 'Internal Server Error' when user closes his last issue)

Rumega committed:
Localization support in mail messages (not tested with Redmine 1.x, but it should work).

ande3577 committed:
Closes #19: Move to sidebar

ande3577 committed:
Closes #15: enable csv export
	Can view others stuff to do
	Filters for available not supported
	PDF support TBD

ande3577 committed:
Closes #20: filter improvements
  no longer requires ability to view others' stuff_to_do
  no longer overrides user selection
  no longer displays closed statuses in the list

ande3577 committed:
Closes #2: Each user has a list of stuff to do reportees
  When viewing own stuff to do, will only see reportees
  Reportees can be set by admin or by individual user (with manage
    reportees permission)
  Can add either a user or a group as reportees

ande3577 committed:
Closes #22: fix access errors for filters.

ande3577 committed:
no longer deletes title bar
admin menu now part of layout
Added links back to user/account settings from reportees page

ande3577 committed:
Closes #23: tweak to back links format

neffets committed:
refs #6551 - merge stuff of "Romega" - stuff_to_do_mailer is working in redmine 1.4 too, because the former sent the text part in a html-body-wrapped, I now use the Mailer function render_multipart. Added german translation. Missing translation of the text-Message, we have Settings.default_language de, but its sending only english text.

neffets committed:
refs #6551 - merge stuff of "Romega" - stuff_to_do_mailer is working in redmine 1.4 too, because the former sent the text part in a html-body-wrapped, I now use the Mailer function render_multipart. Added german translation. Missing translation of the text-Message, we have Settings.default_language de, but its sending only english text.

mrgcohen commented:
i'm getting this error. thanks for any help!

Started GET "/settings/plugin/stuff_to_do_plugin" for 10.113.19.89 at Thu Jan 31 09:58:14 -0500 2013
Processing by SettingsController#plugin as HTML
  Parameters: {"id"=>"stuff_to_do_plugin"}
  Rendered settings/plugin.html.erb within layouts/admin (5.3ms)
Completed 500 Internal Server Error in 11ms

ActionView::Template::Error (Missing partial settings/stuff_to_do_settings with

ande3577 committed:
Closes #29: add option to export description to csv

neffets committed:
1. show language select in recommend_below e-mail
2. form_tag in Rails 3 gives the form back, does not output directly, so catch it up and output it

neffets committed:
version changed to 0.5.0 because rails3 supported

neffets committed:
README extended for the rake commands in redmine2.x / rails3

neffets committed:
README list formatting only

neffets committed:
README refrence of new commiters and download urls

ande3577 committed:
Closes #18: fix mailer server error

ande3577 committed:
Closes #28: add a second drop-down control to select project

ande3577 committed:
Closes #31: Fix priority enumeration in stylesheet

ande3577 committed:
Closes #32: show issues assigned to group

ande3577 committed:
Closes #32: Allows adding from issue/context menu if assigned to group
  (Only if current user belongs to assigned group)

neffets committed:
Make "time_grid" feature not crash. Changed all link_remote_to to link_to and all labelled_tabular_form_for to labelled_form_for

ande3577 committed:
Closes #33: reordering list resets project filter

neffets committed:
fixes mailer issue, no mails where sent

neffets committed:
Add permission view_others_stuff_to_do, so that privileged user can view and change , not only admins

neffets committed:
translations now work in redmine 2.3, we have not to hardcode the title anymore.

neffets committed:
Forms in rails3 gives have a return value instead of echoing the output.

neffets committed:
Improves filter - normal user can now filter to (status and priority) - no more closed stati are shown

neffets committed:
line for postgres support added, but not yet activated

neffets committed:
bugfix in case statement
utf-8 fix for author name in init.rb

neffets committed:
bugfix: partly reverts @94247f2

neffets committed:
bugfix: js prototype function last() for array does not exists in jquery, using now simple js-array-ops

ande3577 committed:
Closes #34: update jquery to fix on IE

ande3577 committed:
Closes #37: add screenshots

ande3577 committed:
Closes #36: allow hiding issues from manager when viewing other's stuff to do

ande3577 committed:
Closes #39: Add localization for daily and running totals

Commit:
Initial commit

Commit:
Added officer_id to issues table

Commit:
The plugin is ready to go

Commit:
Added example cron.sh file to be used by cronjob

Commit:
SQL statement did not work with MySQL Server. The 'IS' was missing

Commit:
Changed url generating functions because the old ones made mistakes
 with http and https

Commit:
Now it is possible to subscribe to the escalation alert mailer in the account settings

Commit:
Mass-assignment problem with CustomField.type solved

ande3577 committed:
Adds indication when adding to another users stuff to do

Commit:
Added missing accept_api_auth function

It was not possible to call this function through the api (as a cronjob or so).
Now it is possible

Commit:
Excludes closed issues

Now closed issues will no longer be added to the escalation mails

JangoSteve committed:
Added statuses_for_stuff_to_do plugin setting.

JangoSteve committed:
Added statuses_for_stuff_to_do plugin setting.

JangoSteve committed:
Added statuses_for_stuff_to_do filter to listed issues in stuff to do page.

JangoSteve committed:
Added statuses_for_stuff_to_do filter to listed issues in stuff to do page.

JangoSteve committed:
Added status to issue partial on stuff to do page.

JangoSteve committed:
Added status to issue partial on stuff to do page.

JangoSteve committed:
Add default of 'all' to list all open statuses (equivalent to current behavior without this new setting).

JangoSteve committed:
Add default of 'all' to list all open statuses (equivalent to current behavior without this new setting).

JangoSteve committed:
Added user periodic mailer to send list of stuff to do to users periodically.

JangoSteve committed:
Added user periodic mailer to send list of stuff to do to users periodically.

JangoSteve committed:
Fixed mailer links to use whole url in issue and project mailer partials.

JangoSteve committed:
Fixed mailer links to use whole url in issue and project mailer partials.

JangoSteve committed:
Remove issues from stuff to do when issues are updated with unavailable status.

JangoSteve committed:
Remove issues from stuff to do when issues are updated with unavailable status.

JangoSteve committed:
Added periodic summary mailer for email configured for plugin email_to setting.

JangoSteve committed:
Merge periodic-user-mailer from ande3577/redmine-stuff-to-do-plugin
commit#6fa7d1e2b48bd2b8dd720105aed04e39bd38d8e2
Added periodic summary mailer for email configured for plugin email_to setting.

ande3577 committed:
Merge remote-tracking branch 'jangosteve/add-status-config'

ande3577 committed:
Merge remote-tracking branch 'jangosteve/periodic-user-mailer'

cforce committed:
Update de.yml

cforce committed:
Update de.yml

ande3577 committed:
Merge pull request #49 from cforce/patch-3

Update de.yml

ande3577 committed:
Set default mailer threshold to -1 (disabled), to prevent error until mailer can be fixed

ande3577 committed:
Set default mailer threshold to -1 (disabled), to prevent error until mailer can be fixed

ande3577 committed:
Fix error if enabled trackers are nil

ande3577 committed:
Fix error if enabled trackers are nil

ande3577 committed:
Remove time grid (didn't work anyway and not sure if it ever really did)

ande3577 committed:
Fix another reference to available trackers

ande3577 committed:
Fix another reference to available trackers

Commit:
Changed plugins name

The plugins name was changed from escalation_alert to redmine_reminder.
Furthermore you now have to subscribe to this reminder and the number of
issues which are mentioned in a mail is now written in the first line of the mail

Commit:
Names with old files deleted

Commit:
Renamed Subscription custom field

Commit:
Entered github url and raised version number

Commit:
Added email configuration hint

Commit:
Some changes made most of this plugin work on redmine 2.3.3

First and most impoertant the directory name was changed to
"stuff_to_do_plugin" because it has to fit the plugin id defined in
init.rb. Otherwise rails won't find it.
Additionally I made some minor changes which were necessary to be
compatible with the new rails and redmine versions.
Last but not least I added translations for the permissions added in
german and english.

Commit:
Plugin works on 2.3.3., too

Commit:
Added rake db:migrate instruction to README

Commit:
Removed call to update function in 2nd migration step

Commit:
Merge branch 'master' of git@github.com:raafael911/redmine_reminder.git

Commit:
Update README.rdoc

Commit:
Fix no subscribers found

Renaming of subscribing column wasn't considered in the controller, so
no subscribers could be found

Commit:
Merge branch 'master' of git@github.com:raafael911/redmine_reminder.git

Commit:
@Threshold variable was not visible in mailer view

Commit:
French translation by pousterlus

ande3577 committed:
Move manage all users' stuff to do to permission instead of user setting

ande3577 committed:
Move manage all users' stuff to do to permission instead of user setting

ande3577 committed:
Add clear function

ande3577 committed:
Add clear function

Commit:
Update README.rdoc

Precised subscription mechanism description

Commit:
Polish translation by pharao45.

ande3577 committed:
Reenable link to CSV format

ande3577 committed:
Reenable link to CSV format

JangoSteve committed:
Updated stuff_to_do page format to be consistent with redmine format and styling.

JangoSteve committed:
Updated stuff_to_do page format to be consistent with redmine format and styling.

Commit:
Switched to rake task

Mail delivery is now triggered by a rake task. Furthermore the database changed so all old subscriptions are lost. They can easily be resubscribed on the my account page.

Commit:
Update fr.yml

Commit:
Update pl.yml

Commit:
Update en.yml

Commit:
Updated README and description in init.rb

Commit:
Custom reminder period configurable per user

Each user can now overwrite the administrators settings of how many days in the reminder mail will summarize.

Commit:
Raised version in init.rb

Commit:
Corrected cron.sh RAILS_ENV

RAILS_ENV variable was not recognized by rails so it is not explicitly set in the rake command

allisson committed:
Update pt-br.yml.

allisson committed:
Update pt-br.yml.

ande3577 committed:
Merge pull request #54 from allisson/master

Update pt-br.yml.

Commit:
Fixed incorrect available status getting

Commit:
Fixed incorrect available status getting

Commit:
Merge branch 'master' of github.com:SyntSupport/redmine-stuff-to-do-plugin

ande3577 committed:
Merge pull request #55 from SyntSupport/master

Incorrect available status

Commit:
Bezüglich #8755:

Hook umbenannt da er von anderem Plugin überschrieben wurde

Commit:
Merge branch 'master' of github.com:raafael911/redmine_reminder

Conflicts:
	cron.sh

Commit:
Compatible to Redmine 3

Commit:
Merge branch 'master' of github.com:raafael911/redmine_reminder

Conflicts:
	.gitignore
	config/locales/de.yml
	config/locales/en.yml
	config/locales/fr.yml
	config/routes.rb
	init.rb
	test/test_helper.rb

Commit:
Rescued fatally wrong commit

meron1122 committed:
Add polish translate

Commit:
Merge pull request #1 from meron1122/polish-translate

Add polish translate

Commit:
fit for rails4 and redmine 3.x

Commit:
Rails 3+4 compatibility for bulk-operations on issues

Commit:
refs #3 - TimeGrid - fixed time_grid

Commit:
refs #3 - bugfix date_field

Commit:
bump version to 0.6.0 rails4 support, redmine 2.x, old redmine 1.x cann work but no longe have to work

Commit:
merge until ande3577/redmine-stuff-to-do-plugin/commit/95aa3b94d2c0deedfa6d8b76b0c9399dcb2f33a7 - Adds Permissions and User-Setting to enable/disable stuff-to-do-view

Commit:
merge until ande3577/redmine-stuff-to-do-plugin/commit/95aa3b94d2c0deedfa6d8b76b0c9399dcb2f33a7 - Adds Permissions and User-Setting to enable/disable stuff-to-do-view

Commit:
refs ande3577/redmine-stuff-to-do-plugin/commit/c739c8da8bd676cbb91b9a68fb0c1458b56006e5 - enable csv export

Commit:
improve filters

Commit:
Closes #2: Each user has a list of stuff to do reportees
  When viewing own stuff to do, will only see reportees
  Reportees can be set by admin or by individual user (with manage
    reportees permission)
  Can add either a user or a group as reportees
  refs ande3577/redmine-stuff-to-do-plugin#2

Commit:
Closes ande3577/redmine-stuff-to-do-plugin#22: fix access errors for filters.

Commit:
no longer deletes title bar  refs ande3577/redmine-stuff-to-do-plugin
admin menu now part of layout
Added links back to user/account settings from reportees page

Commit:
Closes ande3577/redmine-stuff-to-do-plugin#23: tweak to back links format

Commit:
Closes ande3577/redmine-stuff-to-do-plugin#29: add option to export description to csv

Commit:
Closes ande3577/redmine-stuff-to-do-plugin#28: add a second drop-down control to select project

Commit:
ande3577/redmine-stuff-to-do-plugin#31: Fix priority enumeration in stylesheet

Commit:
ande3577/redmine-stuff-to-do-plugin#32: show issues assigned to group

Commit:
ande3577/redmine-stuff-to-do-plugin#32: Allows adding from issue/context menu if assigned to group
  (Only if current user belongs to assigned group)

Commit:
ande3577/redmine-stuff-to-do-plugin#33: reordering list resets project filter

Commit:
ande3577/redmine-stuff-to-do-plugin#34: update jquery to fix on IE

Commit:
Closes #37: add screenshots

Commit:
ande3577/redmine-stuff-to-do-plugin#36: allow hiding issues from manager when viewing other's stuff to do

Commit:
ande3577/redmine-stuff-to-do-plugin#39: Add localization for daily and running totals

Commit:
Adds indication when adding to another users stuff to do

Commit:
iCSV-Link works. (the changes for removing time-grid I did not follow;
time-gris works for me)

Commit:
Update pt-br.yml.

Commit:
JohnBat26 коррекция

Commit:
JohnBat26/stuff_to_do_plugin commit#e19d0c1cafb6d8a93f03947da688d5abfbd6e6c0

Commit:
JohnBat27/stuff_to_do_plugin 15 May 2015 6e8e3dbc25eb926ceb0f050a336214271f2a6c2b

Commit:
JohnBat27/stuff_to_do_plugin 15 May 2015 6e8e3dbc25eb926ceb0f050a336214271f2a6c2b

Commit:
fixed jQuery 1.10 missing browser function for contextMenu

Commit:
rails3+4 @alectower - Update migration query method calls

Commit:
polish translation thanx to @wojts

Commit:
Merge branch 'master' of https://github.com/ande3577/redmine-stuff-to-do-plugin into ande3577-master
  find_by_... => find_by()

Commit:
Merge branch 'ande3577-master' into merge-other-develops
  find_by_... => find_by()

Commit:
Replaced function call which was incompatible to new Rails version

Commit:
last authors included stuff

Commit:
start for test cases

Dejv6 committed:
add due date, update czech translation

Dejv6 committed:
fix styles for header

Dejv6 committed:
update czech translation in sidebar

Commit:
Bug fixed #5 Shortcuts on Issue-Page with patch by @tknaller

Commit:
relayouting s-t-d javascript and whitespace-changes, move init-timegrid into js-file on domready

Commit:
bug fixed #6 - Time-Grid addItem now working

Commit:
bumping version to 0.6.2

Dejv6 committed:
fix available statuses

Available statuses from settings didn´t work properly

Commit:
fixed #8 - compatibility - make stuff_to_do plugin more ready for forecoming redmine 4, require solely acts_as_list for tableau, because it will be removed from redmine core

Commit:
refs #8 - gemspec worried old redmine

Commit:
layout fix - localized variants of hours and date often to long for progressbar

Commit:
fixed #9 - added issue context_menu for better administer

Commit:
fixed #9 - css overflow made scroll-x (overdue-icons), fixed

Commit:
fixed #9 - z-index move l_hours behind context_menu

Commit:
refs #9 - context_menu without parameters only from version 3.4.X

Commit:
refs #9 - context_menu via StuffToDoHelper for not get conflicts with same function from Redmine itself, which is private

Commit:
Compatibility for 3.4.4

Commit:
preparing for rails5, al is running at rails4 too (redmine 3.4.11), refs #11

Commit:
refs #11, remove deprecated only_path attribute

Commit:
refs #11 - repair tests for rails4 before upgrade

Commit:
refs #11 - some more tests before upgrade

Commit:
Merge branch 'master' of github.com:neffets/redmine-stuff-to-do-plugin

Commit:
dropping old rails2 support, require minimum redmine 3.x

Commit:
change to GPL v3

Commit:
removed old r2 lang translations

Commit:
removing old r2 route definition

Commit:
rails migration, remaining stuff to do the mailings

Commit:
unnecessary arra to_hash and rails/init removed; now running on rails5 5.2

Commit:
Bugfix CSV export

Commit:
gem not yet, record_helper_tags comes only in rails5 so in redmine4

Commit:
prepare for merging, updates czech translations

Commit:
Merge remote-tracking branch 'raafael911/master' into merge-raafael911
some formating, some translations

Commit:
ws changes

Commit:
ws changes, and cleanup translations

Commit:
Merge branch 'compatible-redmine3'

Commit:
bugfix available statuses

Commit:
Merge branch 'compatible-redmine3'
Bugfix available statuses

Commit:
new firefox/redmine 4.1.0 left/right now uses display=flex

mwaddell committed:
updated styling to match latest version of public/themes/alternate/stylesheets/application.css from Redmine

mwaddell committed:
removing superfluous spaces

neffets committed:
Merge pull request #15 from pangaeatech/FixingStyle

Updating stylesheet based on latest redmine alternate

mwaddell committed:
Don't return deleted items

neffets committed:
Merge pull request #16 from pangaeatech/Task15443-DeleteCrash

BUGFIX: Filter out deleted items in controller

Commit:
prefer highest priority before created

Commit:
adding gemfile group

Commit:
fix email empty caused split error

Commit:
add threshold_cnt for stuff_to_do max limit(default 5), and add locale zh-CN, thx to Jiangshan00001

Commit:
Merge branch 'master' of github.com:neffets/redmine-stuff-to-do-plugin

Commit:
fix the reorder.js 422 error bug

Thanx to @Jiangshan00001
when using chrome to access the stuff_to_do.
after drag one issue from available to doing.
click one doing issue and right-menu-> change status from new->doing.
the back_url is reorder.js, so the page will be error.

Commit:
Rails-6 compatibility

kristapsk committed:
Fix links

Link to Little Stream Software git is broken and Redmine Plugins page has changed URL.

neffets committed:
Merge pull request #20 from RTBC-SIA/kristapsk-patch-1

Fix links

neffets committed:
Update README.md https links
